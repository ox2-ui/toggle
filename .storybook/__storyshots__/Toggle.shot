exports[`blue`] = `
<label
  className="dEWRtq"
  style={undefined}>
  <div
    className="cFNsmu"
    disabled={undefined}
    style={undefined}>
    Show profile settings
  </div>
  <div
    className="ddmShN">
    <input
      checked={true}
      className="KrWol"
      color="#3498db"
      disabled={undefined}
      onChange={[Function]}
      type="checkbox" />
    <span
      className="iZgbnP">
      <i
        className="bqdnEt" />
    </span>
  </div>
</label>
`;

exports[`disabled`] = `
<label
  className="dEWRtq"
  style={undefined}>
  <div
    className="ckRhtA"
    disabled={true}
    style={undefined}>
    Show profile settings
  </div>
  <div
    className="ddmShN">
    <input
      checked={true}
      className="KrWol"
      color="#3498db"
      disabled={true}
      onChange={[Function]}
      type="checkbox" />
    <span
      className="iZgbnP">
      <i
        className="bqdnEt" />
    </span>
  </div>
</label>
`;

exports[`in small container`] = `
<div
  style={
    Object {
      "backgroundColor": "GHOSTWHITE",
      "width": "100px",
    }
  }>
  <label
    className="dEWRtq"
    style={undefined}>
    <div
      className="ddmShN">
      <input
        checked={true}
        className="KrWol"
        color="#3498db"
        disabled={undefined}
        onChange={[Function]}
        type="checkbox" />
      <span
        className="iZgbnP">
        <i
          className="bqdnEt" />
      </span>
    </div>
  </label>
</div>
`;

exports[`no label`] = `
<label
  className="dEWRtq"
  style={undefined}>
  <div
    className="ddmShN">
    <input
      checked={true}
      className="KrWol"
      color="#3498db"
      disabled={undefined}
      onChange={[Function]}
      type="checkbox" />
    <span
      className="iZgbnP">
      <i
        className="bqdnEt" />
    </span>
  </div>
</label>
`;

exports[`off`] = `
<label
  className="dEWRtq"
  style={undefined}>
  <div
    className="cFNsmu"
    disabled={undefined}
    style={undefined}>
    Show profile settings
  </div>
  <div
    className="ddmShN">
    <input
      checked={false}
      className="lmkVLy"
      color="#2ecc71"
      disabled={undefined}
      onChange={[Function]}
      type="checkbox" />
    <span
      className="iZgbnP">
      <i
        className="bqdnEt" />
    </span>
  </div>
</label>
`;

exports[`on`] = `
<label
  className="dEWRtq"
  style={undefined}>
  <div
    className="cFNsmu"
    disabled={undefined}
    style={undefined}>
    Show profile settings
  </div>
  <div
    className="ddmShN">
    <input
      checked={true}
      className="lmkVLy"
      color="#2ecc71"
      disabled={undefined}
      onChange={[Function]}
      type="checkbox" />
    <span
      className="iZgbnP">
      <i
        className="bqdnEt" />
    </span>
  </div>
</label>
`;
