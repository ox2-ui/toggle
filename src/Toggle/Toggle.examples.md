### Toggle


On:

    const handleChange = (value) => setState({ mySetting: value });
    initialState = { mySetting: true };


    <Toggle
      value={state.mySetting}
      label={'My label'}
      onChange={handleChange}
    />

Off:

    const handleChange = (value) => setState({ mySetting: value });
    initialState = { mySetting: false };


    <Toggle
      value={state.mySetting}
      label={'My label'}
      onChange={handleChange}
    />

Disabled:

    const handleChange = (value) => setState({ mySetting: value });
    initialState = { mySetting: true };


    <Toggle
      value={state.mySetting}
      disabled={true}
      label={'My label'}
      onChange={handleChange}
    />

Custom color:

    const handleChange = (value) => setState({ mySetting: value });
    initialState = { mySetting: true };


    <Toggle
      value={state.mySetting}
      color={'#3498db'}
      label={'My label'}
      onChange={handleChange}
    />

No label:

    const handleChange = (value) => setState({ mySetting: value });

    <Toggle
      value={state.mySetting}
      value={state.mySetting}
      onChange={handleChange}
    />

Uncontrolled:

    const handleChange = (value) => console.log('üçá', value);

    <Toggle
      label={'My label'}
      onChange={handleChange}
    />

