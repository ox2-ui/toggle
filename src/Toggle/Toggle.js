import React, { PropTypes } from 'react';
import styled from 'styled-components';

const Label = styled.label`
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  user-select: none;
  display: flex;
  justify-content: space-between;
  align-items: center;
  cursor: pointer;
  padding: 3px 7px;
`;

const Text = styled.div`
  font-family: 'Roboto Condensed';
  font-size: 18px;
  color: #1b160e;
  opacity: ${props => props.disabled ? '0.5' : '1'}
`;

const Wrapper = styled.div`
  position: relative;
  pointer-events: auto;
  margin: -5px;
  padding: 5px;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
`;

const Input = styled.input`
  display: none;

  &:checked + span {
    border-color: ${props => props.color};
    background-color: ${props => props.color};
  }

  &:disabled + span {
    opacity: 0.4;
  }

  &:checked + span i {
    transform: translate3d(22px, 0, 0);
    background-color: #fff;
  }
`;

const Track = styled.span`
  transition-timing-function: ease-in-out;
  transition-duration: 0.2s;
  transition-property: background-color, border;
  display: inline-block;
  box-sizing: border-box;
  width: 54px;
  height: 32px;
  border: solid 2px #E5E5E5;
  border-radius: 20px;
  background-color: #E5E5E5;
  content: ' ';
  cursor: pointer;
  pointer-events: none;
`;


const Handle = styled.i`
  transition: 0.2s ease-in-out;
  position: absolute;
  display: block;
  width: 28px;
  height: 28px;
  border-radius: 28px;
  background-color: #fff;
  top: 7px;
  left: 7px;

  &:before {
    position: absolute;
    top: -4px;
    left: -22px;
    padding: 19px 35px;
    content: " ";
  }
`;


/**
 * Toggle Component
 */
const Toggle = (props) => {
  const { color, className, style, label, labelStyle, disabled, onChange, value } = props;

  return (
    <Label style={style} className={className}>
      { label &&
        <Text style={labelStyle} disabled={disabled}>{label}</Text>
      }
      <Wrapper>
        <Input
          type="checkbox"
          color={color || '#2ecc71'}
          disabled={disabled}
          checked={value ? true : false } // eslint-disable-line no-unneeded-ternary
          onChange={(event) => onChange(event.target.checked)} // eslint-disable-line react/jsx-handler-names
        />
        <Track>
          <Handle />
        </Track>
      </Wrapper>
    </Label>
  );
};

Toggle.propTypes = {
  /**
   * ClassName of the root element
   */
  className: PropTypes.string,
  /**
   * Color of the toggle button
   */
  color: PropTypes.string,
  /**
   * Will disable the toggle if true.
   */
  disabled: PropTypes.string,
  /**
   * Label for toggle.
   */
  label: PropTypes.string,
  /**
   * Overrides the inline-styles of the Toggle element label.
   */
  labelStyle: PropTypes.string,
  /**
   * Callback function that is fired when the toggle switch is toggled.
   */
  onChange: PropTypes.string,
  /**
   * Override the inline-styles of the root element.
   */
  style: PropTypes.string,
  /**
   * Value for when using controlled toggle.
   */
  value: PropTypes.string,
};

export default Toggle;
