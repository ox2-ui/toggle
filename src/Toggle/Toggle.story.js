import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Toggle from './Toggle';

const onChange = (value) => console.log('🍇', value); // eslint-disable-line no-console

const style = {
  container: {
    width: '100px',
    backgroundColor: 'GHOSTWHITE',
  },
};

storiesOf('Toggle', module)
  .add('on', () => (
    <Toggle
      value={true}
      label={'Show profile settings'}
      onChange={onChange}
    />
  ))
  .add('off', () => (
    <Toggle
      value={false}
      label={'Show profile settings'}
      onChange={onChange}
    />
  ))
  .add('blue', () => (
    <Toggle
      color={'#3498db'}
      value={true}
      label={'Show profile settings'}
      onChange={onChange}
    />
  ))
  .add('disabled', () => (
    <Toggle
      color={'#3498db'}
      value={true}
      disabled={true}
      label={'Show profile settings'}
      onChange={onChange}
    />
  ))
  .add('no label', () => (
    <Toggle
      color={'#3498db'}
      value={true}
      onChange={onChange}
    />
  ))
  .add('in small container', () => (
    <div style={style.container}>
      <Toggle
        color={'#3498db'}
        value={true}
        onChange={onChange}
      />
    </div>

  ));
