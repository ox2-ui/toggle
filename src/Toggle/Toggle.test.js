/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Toggle from './Toggle';

it('renders enabled', () => {
  const wrapper = mount(
    <Toggle
      value={true}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('reders disabled', () => {
  const wrapper = mount(
    <Toggle
      value={false}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('reders with label', () => {
  const wrapper = mount(
    <Toggle
      value={true}
      label={'My label'}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('reders custom color', () => {
  const wrapper = mount(
    <Toggle
      value={true}
      color={'#3498db'}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
