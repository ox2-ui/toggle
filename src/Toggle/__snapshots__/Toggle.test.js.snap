exports[`test reders custom color 1`] = `
<Toggle
  color="#3498db"
  value={true}>
  <styled.label>
    <label
      className="dEWRtq">
      <styled.div>
        <div
          className="ddmShN">
          <styled.input
            checked={true}
            color="#3498db"
            onChange={[Function]}
            type="checkbox">
            <input
              checked={true}
              className="KrWol"
              color="#3498db"
              onChange={[Function]}
              type="checkbox" />
          </styled.input>
          <styled.span>
            <span
              className="iZgbnP">
              <styled.i>
                <i
                  className="bqdnEt" />
              </styled.i>
            </span>
          </styled.span>
        </div>
      </styled.div>
    </label>
  </styled.label>
</Toggle>
`;

exports[`test reders disabled 1`] = `
<Toggle
  value={false}>
  <styled.label>
    <label
      className="dEWRtq">
      <styled.div>
        <div
          className="ddmShN">
          <styled.input
            checked={false}
            color="#2ecc71"
            onChange={[Function]}
            type="checkbox">
            <input
              checked={false}
              className="lmkVLy"
              color="#2ecc71"
              onChange={[Function]}
              type="checkbox" />
          </styled.input>
          <styled.span>
            <span
              className="iZgbnP">
              <styled.i>
                <i
                  className="bqdnEt" />
              </styled.i>
            </span>
          </styled.span>
        </div>
      </styled.div>
    </label>
  </styled.label>
</Toggle>
`;

exports[`test reders with label 1`] = `
<Toggle
  label="My label"
  value={true}>
  <styled.label>
    <label
      className="dEWRtq">
      <styled.div>
        <div
          className="cFNsmu">
          My label
        </div>
      </styled.div>
      <styled.div>
        <div
          className="ddmShN">
          <styled.input
            checked={true}
            color="#2ecc71"
            onChange={[Function]}
            type="checkbox">
            <input
              checked={true}
              className="lmkVLy"
              color="#2ecc71"
              onChange={[Function]}
              type="checkbox" />
          </styled.input>
          <styled.span>
            <span
              className="iZgbnP">
              <styled.i>
                <i
                  className="bqdnEt" />
              </styled.i>
            </span>
          </styled.span>
        </div>
      </styled.div>
    </label>
  </styled.label>
</Toggle>
`;

exports[`test renders enabled 1`] = `
<Toggle
  value={true}>
  <styled.label>
    <label
      className="dEWRtq">
      <styled.div>
        <div
          className="ddmShN">
          <styled.input
            checked={true}
            color="#2ecc71"
            onChange={[Function]}
            type="checkbox">
            <input
              checked={true}
              className="lmkVLy"
              color="#2ecc71"
              onChange={[Function]}
              type="checkbox" />
          </styled.input>
          <styled.span>
            <span
              className="iZgbnP">
              <styled.i>
                <i
                  className="bqdnEt" />
              </styled.i>
            </span>
          </styled.span>
        </div>
      </styled.div>
    </label>
  </styled.label>
</Toggle>
`;
